packages/gatsby/src/redux/__tests__/run-sift.js

```diff
@@ -64,280 +83,380 @@ if (!process.env.GATSBY_DB_NODES || process.env.GATSBY_DB_NODES === `redux`) {
-                    type: new GraphQLList(
-                      new GraphQLObjectType({
-                        name: `Second`,
+  const typeName = `test`
+  const gqlType = new GraphQLObjectType({
+    name: typeName,
+    fields: () => {
+      return {
+        id: { type: new GraphQLNonNull(GraphQLID) },
+        string: { type: GraphQLString },
+        first: {
+          type: new GraphQLObjectType({
+            name: `First`,
+            fields: {
+              willBeResolved: {
+                type: GraphQLString,
+                resolve: () => `resolvedValue`,
+              },
+              second: {
+                type: new GraphQLList(
+                  new GraphQLObjectType({
+                    name: `Second`,
+                    fields: {
+                      willBeResolved: {
+                        type: GraphQLString,
+                        resolve: () => `resolvedValue`,
+                      },
+                      third: new GraphQLObjectType({
+                        name: `Third`,
                         fields: {
-                          willBeResolved: {
-                            type: GraphQLString,
-                            resolve: () => `resolvedValue`,
-                          },
-                          third: new GraphQLObjectType({
-                            name: `Third`,
-                            fields: {
-                              foo: GraphQLString,
-                            },
-                          }),
+                          foo: GraphQLString,
                         },
-                      })
-                    ),
-                  },
-                },
-              }),
+                      }),
+                    },
+                  })
+                ),
+              },
             },
-          }
+          }),
         },
-      })
-      describe(`filters by just id correctly`, () => {
-        it(`eq operator`, async () => {
-          const queryArgs = {
-            filter: {
-              id: { eq: `id_2` },
-            },
-          }
-
-          const resultSingular = await runSift({
-            gqlType,
-            queryArgs,
-            firstOnly: true,
-            nodeTypeNames: [gqlType.name],
-            typedKeyValueIndexes: createIndexCache(),
-          })
+      }
+    },
+  })
 
-          const resultMany = await runSift({
-            gqlType,
-            queryArgs,
-            firstOnly: false,
-            nodeTypeNames: [gqlType.name],
-            typedKeyValueIndexes: createIndexCache(),
-          })
+  describe(`run-sift tests`, () => {
+    beforeEach(() => {
+      store.dispatch({ type: `DELETE_CACHE` })
+      mockNodes().forEach(node =>
+        actions.createNode(node, { name: `test` })(store.dispatch)
+      )
+    })
+    ;[
+      { desc: `with cache`, cb: () => new Map() }, // Avoids sift for flat filters
+      { desc: `no cache`, cb: () => undefined }, // Always goes through sift
```

Comment:
```suggestion
      { desc: `no cache`, cb: () => null }, // Always goes through sift
```
@pvdz This should be `null` here as well, right?

Reply:
Yeah good catch